resin/settings.py

```diff
@@ -35,15 +36,20 @@ class Settings(object):
                                             'cache')
 
     def __init__(self):
-        if Path.isdir(self._setting['data_directory']):
+        try:
+            self.__read_settings()
+        except:
+            # move old settings file if it exists.
             if Path.isfile(Path.join(self._setting['data_directory'],
                                      self.CONFIG_FILENAME)):
-                self.__read_settings()
-            else:
-                self.__write_settings()
-        else:
-            # data directory doesn't exist
+                shutil.move(Path.join(self._setting['data_directory'],
```

Comment:
@dfunckt I've noticed about this, even a case when we don't have the permissions to write the default settings file again but  there will be an exception thrown in both ways (InvalidSettings by us or OSError as you mention) So I don't think we need this catch here